### PR TITLE
Added Dijkstra Algorithm in C#

### LINKED ISSUE #532: Dijkstra Algorithm in C#
This is an open issue for the Dijkstra Algorithm algorithm implementation in C#. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by riturajjain2000**
**Fixes #532**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,6 +146,8 @@ The community maintained a list of **Algorithms and Data Structures** implementa
       <td><code>gr-dij</code></td>
       <td class="text-center"></td>
       <td class="text-center">&#10004;</td>
+      <td class="text-center">&#10004;</td>
+      <td class="text-center">&#10004;</td>
```
Comment (Danushka96) [inline README.md:150]: Hi @riturajjain2000 
Why two languages are ticked in here?

Comment (riturajjain2000) [reply README.md:150]: Oh ,sorry I added it by mistake, removed the extra line

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	} 
+
+	// Function to print the constructed distance array 
+	static void printSolution(int[] dist) 
+	{ 
+		Console.Write("Vertex \t\t Distance "
+					+ "from Source\n"); 
+		for (int i = 0; i < V; i++) 
+		{
+			Console.Write(i + " \t\t " + dist[i] + "\n");
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:32]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			} 
+		}
+
+		return min_index; 
+	} 
+
+	// Function to print the constructed distance array 
+	static void printSolution(int[] dist) 
+	{ 
+		Console.Write("Vertex \t\t Distance "
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:28]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
+// A C# program for Dijkstra's  shortest path algorithm. 
+using System; 
+
+class Dijkstra {
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:4]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+		{ 
+			
+			int u = minDistance(dist, sptSet); 
+
+			// Mark the picked vertex as processed 
+			sptSet[u] = true; 
+
+			// Update dist value of the adjacent 
+			// vertices of the picked vertex. 
+			for (int v = 0; v < V; v++)
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:65]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+			// Mark the picked vertex as processed 
+			sptSet[u] = true; 
+
+			// Update dist value of the adjacent 
+			// vertices of the picked vertex. 
+			for (int v = 0; v < V; v++) 
+
+		
+				if (!sptSet[v] && graph[u, v] != 0 && dist[u] != int.MaxValue && dist[u] + graph[u, v] < dist[v])
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:68]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs (hunk 5 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:65]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs (hunk 2 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:32]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs (hunk 6 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:68]: Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs (hunk 3 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:28]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs (hunk 4 above)
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:4]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+				min = dist[v]; 
+				min_index = v; 
+			} 
+		}
+
+		return min_index; 
+	} 
+
+	// Function to print the constructed distance array 
+	static void printSolution(int[] dist)
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:26]: Codacy found an issue: [Remove this unused method parameter 'dist'.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+		return min_index; 
+	} 
+
+	// Function to print the constructed distance array 
+	static void printSolution(int[] dist) 
+	{ 
+		
+		for (int i = 0; i < V; i++) 
+		{
+			Console.Write(dist[i]);
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:31]: Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

File: algorithms/gr-dij/csharp/DijkstraAlgorithm.cs [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+				if (!sptSet[v] && graph[u, v] != 0 && dist[u] != int.MaxValue && dist[u] + graph[u, v] < dist[v]) {
+					dist[v] = dist[u] + graph[u, v]; 
+				}
+			}
+		} 
+
+		// print the constructed distance array 
+		for (int i = 0; i < V; i++) 
+		{
+			//Console.Write(dist[i]);
```
Comment (RavinduSachintha) [inline algorithms/gr-dij/csharp/DijkstraAlgorithm.cs:68]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6349348)

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
Hi @riturajjain2000 
Thanks for the contribution, keep it up!

**Pull request merged**